File: guides/micronaut-mqtt/cli/src/main/java/example/micronaut/TemperatureClient.java
package example.micronaut;

import io.micronaut.mqtt.annotation.Topic;
import io.micronaut.mqtt.v5.annotation.MqttPublisher;
import java.math.BigDecimal;

@MqttPublisher // <1>
public interface TemperatureClient {

    @Topic("house/livingroom/temperature") // <1>
    void publishLivingroomTemperature(byte[] data);
}

File: guides/micronaut-mqtt/cli/kotlin/src/main/kotlin/example/micronaut/TemperatureClient.kt
package example.micronaut

import io.micronaut.mqtt.annotation.Topic
import io.micronaut.mqtt.annotation.v5.MqttPublisher

@MqttPublisher // <1>
interface TemperatureClient {

    @Topic("house/livingroom/temperature") // <2>
    fun publishLivingroomTemperature(data: ByteArray)
}
